src/index.ts

```diff
@@ -351,10 +356,16 @@ export class UnleashClient extends TinyEmitter {
                     headers: this.getHeaders(),
                     body,
                 });
+
                 if (response.ok && response.status !== 304) {
-                    this.etag = response.headers.get('ETag') || '';
+                    const etag = response.headers.get('ETag');
                     const data = await response.json();
-                    await this.storeToggles(data.toggles);
+                    
+                    if(etag && etag !== this.etag) {
+                        await this.storeToggles(data.toggles, etag);
+                    } else if(!etag) {
```

Comment:
Needed to support the case where a proxy do not fancy to send 304 responses at all.